packages/veraswap-sdk/src/calls/getBridgeSwapCalls.ts

```diff
@@ -27,11 +56,202 @@ export interface GetBridgeSwapCallsParams extends GetCallsParams {
  * @param _queryClient
  * @param _wagmiConfig
  */
-export function getBridgeSwapCalls(
+export async function getBridgeSwapCalls(
     queryClient: QueryClient,
     wagmiConfig: Config,
     params: GetBridgeSwapCallsParams,
 ): Promise<GetCallsReturnType> {
-    //TODO: Add swap calls
-    return getTransferRemoteWithFunderCalls(queryClient, wagmiConfig, params);
+    const {
+        amount,
+        destination,
+        hook,
+        hookMetadata,
+        token,
+        chainId,
+        account,
+        originERC7579ExecutorRouter,
+        remoteERC7579ExecutorRouter,
+        remoteSwapParams,
+        tokenStandard,
+        approveAmount,
+        permit2,
+    } = params;
+    const { initData, salt } = params.createAccount;
+    const contracts = params.contracts ?? {
+        execute: LOCAL_KERNEL_CONTRACTS.execute,
+        ownableSignatureExecutor: LOCAL_KERNEL_CONTRACTS.ownableSignatureExecutor,
+    };
+
+    if (!(tokenStandard === "HypERC20" || tokenStandard === "HypERC20Collateral" || tokenStandard === "NativeToken")) {
+        throw new Error(`Unsupported standard for bridge and swap: ${tokenStandard}`);
+    }
+
+    // Create account if needed
+    const createAccountCalls = await getKernelFactoryCreateAccountCalls(queryClient, wagmiConfig, {
```

Comment:
Need to check remote chain for account creation as well

Reply:
Edit: technically we can always pass the init data in the hyperlane message but cleaner if we dont.